src/test/ui/structs-enums/enum-non-c-like-repr-int.rs

```diff
@@ -65,8 +65,11 @@ fn main() {
     unsafe {
         // This should be safe, because we don't match on it unless it's fully formed,
         // and it doesn't have a destructor.
-        #[allow(deprecated)]
-        let mut dest: MyEnum = mem::uninitialized();
+        //
+        // Furthermore, there are no types within MyEnum which cannot be initialized with zero,
+        // specifically, though padding and such are present, there are no references or similar
+        // types.
```

Comment:
Well on top of that we also need to know that 0 is a valid discriminant. Not sure if that is guaranteed?

Reply:
Hm, yes, true. I think we do at least intend to guarantee this.

> As in C, discriminant values that are not specified are defined as either 0 (for the first variant) or as one more than the prior variant.

from https://rust-lang.github.io/unsafe-code-guidelines/layout/enums.html

and also we have 

>  If the first enumerator does not have an initializer, the associated value is zero. For any other enumerator whose definition does not have an initializer, the associated value is the value of the previous enumerator plus one. 

from https://en.cppreference.com/w/cpp/language/enum

Reply:
The UCG quote is inside a section on enums with a `repr` though, isn't it?

And what C++ does is entirely insignificant.

Reply:
Hm, my impression is that C++ tries to be compatible to C in this regard, though I was unable to quickly find C specification that matches this -- since these enums are repr(C) I presumed that would help...

As far as I can tell, this test case seems to match [this section](https://rust-lang.github.io/unsafe-code-guidelines/layout/enums.html#explicit-repr-annotation-with-c-compatibility) of the UCG docs -- i.e., the discriminant is "just" a byte...

Ideally I think what this wants is `MaybeUninit` with field-level initialization, but unfortunately we just don't have that today :/

Reply:
Oh, I missed that these enums have a `repr`. Then we should be good, as per the UCG document you references. But the comment should mention the `repr`.

Reply:
Will update to mention the repr. Thanks!